### PR TITLE
fix(shared, nextjs): Support importing @clerk/shared in @clerk/nextjs

### PR DESCRIPTION
**Pull request by dimkl**
## Description

**Review it per commit**

Use `use client` directive in @clerk/shared to support importing it in libraries compatible with RSC and use package in @clerk/nextjs (also replaced current copy/pasted code).

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [x] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/nextjs/src/server/utils.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import type { RequestState } from '@clerk/backend';
 import { buildRequestUrl, constants } from '@clerk/backend';
+import { handleValueOrFn, isHttpOrHttps } from '@clerk/shared';
```
Comment (panteliselef) [inline packages/nextjs/src/server/utils.ts:3]: 🎉🎉🎉

**Review by panteliselef — APPROVED** [review]
Do wee need to test this with snapshots ? Or should we be good as long this builds ? I remember someone mentioned a runtime error that is why i am asking.

**Comment by panteliselef** [comment]
Sorry didn't mean to approve, just to add a comment.

**Comment by dimkl** [comment]
@panteliselef do know mean this runtime error ?: https://github.com/clerkinc/javascript/pull/1760#discussion_r1333464629

**Comment by panteliselef** [comment]
> @panteliselef do know mean this runtime error ?: [#1760 (comment)](https://github.com/clerkinc/javascript/pull/1760#discussion_r1333464629)

yes, that one

**Review by anagstef — APPROVED** [review]

File: .changeset/unlucky-emus-greet.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@clerk/nextjs': minor
+'@clerk/shared': minor
+---
+
+Use `use client` directive in @clerk/shared to support import in libraries compatible in RSC and use it in @clerk/nextjs
```
Comment (BRKalow) [inline .changeset/unlucky-emus-greet.md:6]:
```suggestion
Add the `use client` directive in `@clerk/shared` to make the package compatible with an RSC environment. Remove several helpers from `@clerk/nextjs` and import them from `@clerk/shared` instead.
```

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**